src/main/java/net/pms/formats/FormatFactory.java

```diff
@@ -148,7 +147,24 @@ public static Format getAssociatedFormat(final String filename) {
 			formatsLock.readLock().unlock();
 		}
 
-		LOGGER.trace("Could not match any format to \"" + filename + "\"");
+		LOGGER.trace("Could not match any format to \"{}\"", filename);
+		return null;
+	}
+
+	public static Format getFormat(Class<? extends Format> clazz) {
+		if (clazz == null) {
+			return null;
+		}
+		formatsLock.readLock().lock();
+		try {
+			for (Format format : formats) {
+				if (format.getClass().equals(clazz)) {
```

Comment:
@Sami32 It's too much to explain here, read about Java Generics if you want to learn how it works. It has nothing to do with ISOVOB, it's a general method.